File: _mock_data/login_users/convert2csv.py
import re
import csv

sql_file = "existing_users.sql"
csv_file = "existing_users.csv"

# Regular expression pattern to match the values in the SQL statements
pattern = r"values \('.*?', '.*?', '.*?', '.*?', '(.*?)'\)"

# Open the SQL file for reading
with open(sql_file, 'r') as file:
    sql_content = file.read()

# Extract the usernames and passwords using regex
matches = re.findall(pattern, sql_content)

# Create a list of dictionaries with the extracted data
data = [{'username': match[0], 'password': match[1]} for match in zip(matches[::2], matches[1::2])]

# Write the data to the CSV file
with open(csv_file, 'w', newline='') as file:
    writer = csv.DictWriter(file, fieldnames=['username', 'password'])
    writer.writeheader()
    writer.writerows(data)

print("Data successfully extracted and written to the CSV file.")

File: _mock_data/login_users/convert2csv.js
const fs = require('fs');
const csv = require('csv-writer').createObjectCsvWriter;

const sqlFile = 'existing_users.sql';
const csvFile = 'existing_users.csv';

// Regular expression pattern to match the values in the SQL statements
const pattern = /values \('.*?', '.*?', '.*?', '.*?', '(.*?)'\)/g;

// Read the SQL file
const sqlContent = fs.readFileSync(sqlFile, 'utf8');

// Extract the usernames and passwords using regex
const matches = [...sqlContent.matchAll(pattern)];

// Create an array of objects with the extracted data
const data = matches.map((match) => ({
  username: match[1],
  password: match[2],
}));

// Write the data to the CSV file
const csvWriter = csv({
  path: csvFile,
  header: [{ id: 'username', title: 'username' }, { id: 'password', title: 'password' }],
});

csvWriter.writeRecords(data)
  .then(() => {
    console.log('Data successfully extracted and written to the CSV file.');
  })
  .catch((error) => {
    console.error('Error writing to the CSV file:', error);
  });
